src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,20 +22,46 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+class FileInfo(val info: String) extends Info {
+  override def toString: String = " @[" + info + "]"
   //scalastyle:off method.name
   def ++(that: Info): Info = if (that == NoInfo) this else MultiInfo(Seq(this, that))
 }
+
+object FileInfo {
+  @deprecated("The use of FileInfo(...) is deprecated. Use FileInfo.fromUnEscaped(...) instead", "1.4")
```

Comment:
Two nits about writing deprecation warnings:

1. This will auto-generate the message about "FileInfo is deprecated" so the PR can leave that off.
2. The PR should include the library name in the "since" parameter.

The above two are somewhat canonical with how Scala does it. In Chisel and FIRRTL, we've tended to adopt also adding a "This will be removed in FIRRTL 1.5" message to be explicit with how long somebody has to get their code updated.

Consider:

```suggestion
  @deprecated("Use FileInfo.fromUnEscaped instead. FileInfo.apply will be removed in FIRRTL 1.5.", "FIRRTL 1.4")
```

I believe that will then generate the following message:

```
method apply in object FileInfo is deprecated (since FIRRTL 1.4): Use FileInfo.fromUnEscaped instead. FileInfo.apply will be removed in FIRRTL 1.5.
```

Reply:
Thanks!